openstack/orchestration/v1/resourcetypes/requests.go

```diff
@@ -0,0 +1,120 @@
+package resourcetypes
+
+import (
+	"fmt"
+
+	"github.com/gophercloud/gophercloud"
+)
+
+// SupportStatus is a type for specifying by which support status to filter the
+// list of resource types.
+type SupportStatus string
+
+var (
+	// SupportStatusUnknown is returned when the resource type does not have a
+	// support status.
+	SupportStatusUnknown SupportStatus = "UNKNOWN"
+	// SupportStatusSupported indicates a resource type that is expected to
+	// work.
+	SupportStatusSupported SupportStatus = "SUPPORTED"
+	// SupportStatusDeprecated indicates a resource type that is in the process
+	// being removed, and may or may not be replaced by something else.
+	SupportStatusDeprecated SupportStatus = "DEPRECATED"
+	// SupportStatusHidden indicates a resource type that has been removed.
+	// Existing stacks that contain resources of this type can still be
+	// deleted or updated to remove the resources, but they may not actually
+	// do anything any more.
+	SupportStatusHidden SupportStatus = "HIDDEN"
+	// SupportStatusUnsupported indicates a resource type that is provided for
+	// preview or other purposes and should not be relied upon.
+	SupportStatusUnsupported SupportStatus = "UNSUPPORTED"
+)
+
+// ListOptsBuilder allows extensions to add additional parameters to the
+// List request.
+type ListOptsBuilder interface {
+	ToResourceTypeListQuery() (string, error)
+	ReturnDescription() bool
+}
+
+// ListOpts allows the filtering of collections through the API.
+type ListOpts struct {
+	// Filters the resource type list by a regex on the name.
+	Pattern string `q:"name"`
```

Comment:
The Field name should match the JSON tag: `Name`.

Reply:
I used a different name because this API has proved confusing to users: it **isn't** checking for that resource type by name, but filtering the list by a regex. I'd change the API if I could, but although that horse has bolted it would be nice to avoid perpetuating the confusion here. Would something like `NameRegEx` be acceptable?